frontend/packages/dev-console/src/components/import/advanced/DeploymentConfigSection.tsx

```diff
@@ -1,19 +1,27 @@
 import * as React from 'react';
+import * as _ from 'lodash';
 import { CheckboxField, EnvironmentField } from '@console/shared';
+import { K8sResourceKind } from '@console/internal/module/k8s';
 import FormSection from '../section/FormSection';
 
 export interface DeploymentConfigSectionProps {
   namespace: string;
+  resource?: K8sResourceKind;
 }
 
-const DeploymentConfigSection: React.FC<DeploymentConfigSectionProps> = ({ namespace }) => {
-  const deploymentConfigObj = {
+const DeploymentConfigSection: React.FC<DeploymentConfigSectionProps> = ({
+  namespace,
+  resource,
+}) => {
+  const deploymentConfigObj = resource || {
     kind: 'DeploymentConfig',
     metadata: {
       namespace,
     },
   };
-
+  const envs = _.get(deploymentConfigObj, 'spec.template.spec.containers[0].env', []).map((env) =>
+    _.values(env),
+  );
```

Comment:
If every client needs to convert an array of objects with a name / value(From) pairs, why not move this into the `EnvironmentField`.
Pass `env` as an array of objects instead?

Reply:
Moved it into EnvironmentField and passing env as array.